internal/webhooks/cluster_test.go

```diff
@@ -1266,18 +1403,28 @@ func TestClusterTopologyValidation(t *testing.T) {
 			// Sets up the fakeClient for the test case.
 			fakeClient := fake.NewClientBuilder().
 				WithObjects(class).
+				WithObjects(tt.additionalObjects...).
 				WithScheme(fakeScheme).
 				Build()
 
+			if tt.clusterCacheTrackerReader == nil {
+				tt.clusterCacheTrackerReader = &fakeClusterCacheTracker{client: fake.NewFakeClient()}
+			}
+
 			// Create the webhook and add the fakeClient as its client. This is required because the test uses a Managed Topology.
-			webhook := &Cluster{Client: fakeClient}
+			webhook := &Cluster{Client: fakeClient, Tracker: tt.clusterCacheTrackerReader}
 
 			warnings, err := webhook.validate(ctx, tt.old, tt.in)
 			if tt.expectErr {
 				g.Expect(err).To(HaveOccurred())
 				g.Expect(warnings).To(BeEmpty())
 				return
 			}
+			if tt.expectWarning {
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(warnings).ToNot(BeEmpty())
+				return
+			}
 			g.Expect(err).ToNot(HaveOccurred())
 			g.Expect(warnings).To(BeEmpty())
```

Comment:
```suggestion
			if tt.expectErr {
				g.Expect(err).To(HaveOccurred())
			} else {
				g.Expect(err).ToNot(HaveOccurred())
			}
			if tt.expectWarning {
				g.Expect(warnings).ToNot(BeEmpty())
			} else {
				g.Expect(warnings).To(BeEmpty())
			}
```

Let's use this instead please. Just easier to think through all the combinations and it's easier if each expect just maps to error or warning